### DISCUSSION TITLE
Customize Bootstrap 5.3 with CSS variables

### DISCUSSION THREAD

**Discussion by manuelmeurer** [twbs/bootstrap#39551] [Q&A] [answered]
I'm starting a new project with Bootstrap 5.3 and I'm trying to do some basic customizations with CSS variables but I'm unsure what the best approach is...

Basically I'm loading the CSS from a CDN, and as a first step I'd like to customize the primary color that is used everywhere.

```
<!DOCTYPE html>
<html>
  <head>
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <link rel="stylesheet" href="https://ga.jspm.io/npm:bootstrap@5.3.2/dist/css/bootstrap.min.css" />
    <style>
      :root,
      [data-bs-theme=light] {
        --bs-primary: yellow;
      }
    </style>
  </head>

  <body>
    <a class="btn btn-primary" href="#">Link</a>
  </body>
</html>
```

This has no effect, though, the button is still displayed in the default primary blue.

What is the correct way to override the primary color with CSS variables?

**Comment by manuelmeurer** [comment]
Ok I found [this article](https://dev.to/lixeletto/4-dead-simple-ways-of-customizing-bootstrap-5132) that says I need to change `--bs-btn-bg` on `.btn-primary`.

Is that correct?
Does that mean there is no simple "one line" fix to change the primary color (or other basic values that are used in a lot of other places) with CSS variables?

Do I need to go back to using Sass, so I can set `$primary`?
I was really hoping I could use the compiled CSS from a CDN and do a few simply customizations with CSS vars... 😄

**Comment by julien-deramond** [comment, marked as answer]
Hey @manuelmeurer 

So right now, tweaking the `--bs-primary` CSS variable to give all components a new color isn't possible. You can see `--bs-primary` primarily as a read-only CSS variable reusable in your website to get the primary color. For a consistent global look and setting a new primary color, diving into Sass is the way to go.

However, as explained in the article, depending on the needs, it's possible to override the rendering of the components one by one by overriding their CSS variables.

Some improvements will be made when Bootstrap is refactored in v6 for an easier theming approach. The groundwork with CSS variables and color modes in v5.2/3 was the opening act, and we committed to keeping things smooth without any regressions.

**Comment by manuelmeurer** [comment]
Thanks for the explanation, @julien-deramond!

I'll stick with Sass for now and look forward to v6! :)

**Comment by cgslivre** [comment]
https://bootstrap.build
